src/components/Staff/ManagePromotion.vue

```diff
@@ -0,0 +1,347 @@
+                      </v-col>
+                      <v-col cols="12" sm="6" md="4">
+                        <v-text-field
+                          required
+                          :rules="rules.percent"
+                          v-model="editedItem.percentoff"
+                          color="orange darken-3"
+                          label="Percent Off (%)"
+                        ></v-text-field>
+                      </v-col>
+                      <v-col cols="12" sm="6" md="4">
+                        <v-text-field
+                          required
+                          prefix="$"
+                          :rules="rules.minSpend"
+                          v-model="editedItem.minspending"
+                          color="orange darken-3"
+                          label="Minimum Spend (SGD)"
+                        ></v-text-field>
+                      </v-col>
+                      <v-col cols="12" sm="6" md="4">
+                        <v-text-field
+                          required
+                          :rules="rules.monthsWithNoOrders"
+                          v-model="editedItem.monthswithnoorders"
+                          color="orange darken-3"
+                          label="Months With No Orders"
+                        ></v-text-field>
+                      </v-col>
+                    </v-row>
+                  </v-container>
+                </v-card-text>
+
+                <v-card-actions>
+                  <v-spacer></v-spacer>
+                  <v-btn color="orange darken-1" text @click="close">Cancel</v-btn>
+                  <v-btn :disabled="!validForm" color="orange darken-1" text @click="save">Save</v-btn>
+                </v-card-actions>
+              </v-form>
+            </v-card>
+          </v-dialog>
+        </v-toolbar>
+      </template>
+      <template v-slot:item.actions="{ item }">
+        <v-icon small class="mr-2" @click="editItem(item)">mdi-pencil</v-icon>
+        <v-icon small @click="deleteItem(item)">mdi-delete</v-icon>
+      </template>
+    </v-data-table>
+  </v-container>
+</template>
+
+<script>
+import axios from "axios";
+export default {
+  data: () => ({
+    rid: null,
+    promoList: [],
+    dialog: false,
+    editedIndex: -1,
+    rangeDates: [],
+    menu: false,
+    dateToday: null,
+    validForm: true,
+    editedItem: {
+      pid: null,
+      startdate: null,
+      enddate: null,
+      points: 0,
+      percentoff: 0,
+      minspending: 0,
+      monthswithnoorders: 0
+    },
+    defaultItem: {
+      pid: null,
+      startdate: null,
+      enddate: null,
+      points: 0,
+      percentoff: 0,
+      minspending: 0,
+      monthswithnoorders: 0
+    },
+    promoHeaders: [
+      { text: "Promo ID", sortable: true, value: "pid" },
+      { text: "Start Date", sortable: true, value: "startdate" },
+      { text: "End Date", sortable: true, value: "enddate" },
+      { text: "Points", sortable: false, value: "points" },
```

Comment:
Just curious, why are the rest not sortable?

Reply:
it's unintended, will change it.